OpenDreamRuntime/Procs/NativeProc.cs

```diff
@@ -0,0 +1,84 @@
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Text;
+using OpenDreamRuntime.Objects;
+using OpenDreamShared.Dream.Procs;
+using OpenDreamRuntime.Procs.Native;
+
+namespace OpenDreamRuntime.Procs {
+    public class NativeProc : DreamProc {
+        public delegate DreamValue HandlerFn(DreamObject src, DreamObject usr, DreamProcArguments arguments);
+
+        public class State : ProcState
+        {
+            public DreamObject Src;
+            public DreamObject Usr;
+            public DreamProcArguments Arguments;
+            
+            private NativeProc _proc;
+            public override DreamProc Proc => _proc;
+
+            public State(NativeProc proc, DreamThread thread, DreamObject src, DreamObject usr, DreamProcArguments arguments)
+                : base(thread)
+            {
+                _proc = proc;
+                Src = src;
+                Usr = usr;
+                Arguments = arguments;
+            }
+
+            protected override ProcStatus InternalResume()
+            {
+                if (DreamProcNativeRoot.CurrentRuntime != null) {
+                    throw new InvalidOperationException();

```

Comment:
This causes an issue with nested native proc calls. An example of one of these is the native proc `image()`, which creates an `/image` object. This calls `/image.New()`, which calls the native proc `istext()`.

Reply:
This `DreamProcNativeRoot.CurrentRuntime` is a bit of a hack I was hoping won't be permanent (but it definitely beats editing every single native proc even more inside this single PR.)

I can change this static var to a stack for now - but ideally that `/image/New` call will happen inside the same DreamThread as the `/image()` proc. This is probably best achieved by changing `image` to an AsyncNativeProc because... that's where all the complex functionality like that lives (even if the proc ends up never yielding.)

Reply:
Currently, the `New()` call on every object creates a new DreamThread. This is because `New()` is called in the meta objects' `OnObjectCreated` using `SpawnProc`.

Reply:
Yeah, that behaviour can stay for now. It's not ideal only because it'll lead to worse stack traces in errors and won't allow New to be asynchronous (is it ever?)

I've fixed all the other issues locally, just working on this last one now.